nanoFramework.Json/TimeExtensions.cs

```diff
@@ -109,15 +121,27 @@ public static DateTime FromIso8601(string date)
 			string second = (parts.Length > 5) ? parts[5] : "0";
 			string ms = (parts.Length > 6) ? parts[6] : "0";
 
+			// If the input string is non-numeric, Convert.ToInt32() below will throw an exception from the Native code
+			// These exceptions cause annoying delays & output when running in the nanoFramework debugger
+			// Bail out if the input string is non-numeric
+			Match match1 = Regex.Match(year, @"^[0-9]*$");
```

Comment:
In case we would use Regex, please also declare the matching as a static compiled element, so we will gain into performance